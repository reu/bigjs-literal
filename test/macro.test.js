const babel = require("@babel/core");

const transform = file =>
  babel
    .transformFileSync(file, {
      babelrc: false,
      plugins: [require.resolve("babel-plugin-macros")],
      filename: __filename,
    })
    .code
    .trim();

describe("bigjs-literal/macro", () => {
  it("transforms basic expressions", () => {
    expect(transform("./test/macro/basic.js")).toMatchSnapshot();
  });

  it("transforms interpolated expressions", () => {
    expect(transform("./test/macro/interpolation.js")).toMatchSnapshot();
  });
});
