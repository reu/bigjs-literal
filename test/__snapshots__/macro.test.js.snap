// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bigjs-literal/macro transforms basic expressions 1`] = `"require(\\"big.js\\")(\\"1.5\\").plus(require(\\"big.js\\")(\\"2.5\\"));"`;

exports[`bigjs-literal/macro transforms interpolated expressions 1`] = `
"const a = 2.5;

require(\\"big.js\\")(\\"1.5\\").plus(require(\\"big.js\\")(a));"
`;

exports[`bigjs-literal/macro transforms recursive interpolated expressions 1`] = `"require(\\"big.js\\")(\\"1.5\\").plus(require(\\"big.js\\")(require(\\"big.js\\")(\\"2.5\\").plus(require(\\"big.js\\")(\\"3.5\\"))));"`;
